Objects/genobject.c

```diff
@@ -254,18 +234,29 @@ gen_send_ex2(PyGenObject *gen, PyObject *arg, PyObject **presult,
         _PyErr_ChainStackItem();
     }
 
-    gen->gi_frame_state = FRAME_EXECUTING;
+#if defined(Py_GIL_DISABLED) && defined(Py_DEBUG)
+    ((_PyThreadStateImpl *)tstate)->gen_last_frame_state = FRAME_EXECUTING;
```

Comment:
why is this set only in debug builds?

Reply:
I changed this to `#if defined(Py_GIL_DISABLED)`. It's not really needed at all, but it'll help catch bugs in case some ceval code forgets to set `gen_last_frame_state` before returning.

In other words, when `_PyEval_EvalFrame` returns, `gen_last_frame_state` must be either `FRAME_SUSPENDED[_YIELD_FROM]` or `FRAME_CLEARED` and this helps differentiate this call to `_PyEval_EvalFrame` from earlier calls in the same thread.